### ISSUE TITLE
Avoid "Caching service dependencies ..." on every boot

### ISSUE THREAD

**Issue by hannob** [OpenRC/openrc#455]
While analyzing my boot speed I noticed that openrc will always regenerate the service dependency cache before booting.

Looking into the code it looks like this is enforced. In src/rc/rc.c in the function do_sysvinit() there's a call "_rc_deptree_load(1, NULL);". The first parameter (1) forces the cache regeneration.

This seems unnecessary. _rc_deptree_load() calls rc_deptree_update_needed(), and if needed will update the dependency cache. I can't really think of a situation where there would be a need to regenerate the service cache, except if there's some very specific corruption of the cache files and their mtimes.

This cache updating takes notable time (~0.5 sec on my laptop, which seems significant given I have around 10 secs boot time overall). It also takes longer the more init scripts are installed, and some of them call external applications (e.g. the apache init script will do several calls to the apache binary to validate the config file). This does not appear like something that should happen on every boot.

I have patched this out on my system and it seems to work without problems, but opening an issue first before opening a PR to have some discussion. Are there things I'm missing that make this necessary?

```
diff -Naur a/src/rc/rc.c b/src/rc/rc.c
--- a/src/rc/rc.c	2021-09-21 18:21:17.000000000 +0200
+++ b/src/rc/rc.c	2021-09-28 18:44:09.270114076 +0200
@@ -497,7 +497,7 @@
 	if ((sys = rc_sys()))
 		setenv("RC_SYS", sys, 1);
 	/* force an update of the dependency tree */
-	if ((main_deptree = _rc_deptree_load(1, NULL)) == NULL)
+	if ((main_deptree = _rc_deptree_load(0, NULL)) == NULL)
 		eerrorx("failed to load deptree");
 }

**Comment by thesamesam** [comment]
> @cromerc This is all happening before the sysinit runlevel, so /var will never be mounted at this point if it is a separate file system from /.

Exactly: optimise it for the monolithic / case where we can but fallback if the cache doesn’t exist.

**Comment by cromerc** [comment]
@williamh Good point, I hadn't thought about that. In this case it would only be a boost for those that don't have /var mounted on a separate partition which would be the case for a majority of users. For example installers like calamares and the like don't make a separate /var partition during install with the standard install and default configuration. For someone to have /var on a separate partition usually means manual partitioning and that they chose to mount /var for some specific reason.

The biggest problem I see is that the data for the cache belongs in /var standards wise, there are not really any other places that fit unless you go against standards. And by sticking it in / that does go against standards.

Maybe a compromise would be to have a setting in rc.conf that specifies the cache location with the default being in /var. That way each distro could decide where the cache should be stored. And if a user has /var on a separate partition then it would just mean they need to modify rc.conf to point to somewhere on the root partition to be able to take advantage of the cache.

**Comment by hannob** [comment]
Ping? I think PR #457 should be merged.
